File: CS/WindowsApplication1/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
                private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }


        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(10);
            gridView1.CalcRowHeight += new RowHeightEventHandler(gridView1_CalcRowHeight);
        }


        int minRowHeigth =20;

        void gridView1_CalcRowHeight(object sender, RowHeightEventArgs e)
        {
            GridViewInfo viewInfo = gridView1.GetViewInfo() as GridViewInfo;
            int totalHeight = viewInfo.ViewRects.Rows.Height - minRowHeigth;
            int targetHeight = totalHeight / gridView1.DataRowCount;
            e.RowHeight = Math.Max(minRowHeigth, targetHeight);
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
				Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
				End Function


		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = CreateTable(10)
			AddHandler gridView1.CalcRowHeight, AddressOf gridView1_CalcRowHeight
		End Sub


		Private minRowHeigth As Integer =20

		Private Sub gridView1_CalcRowHeight(ByVal sender As Object, ByVal e As RowHeightEventArgs)
			Dim viewInfo As GridViewInfo = TryCast(gridView1.GetViewInfo(), GridViewInfo)
			Dim totalHeight As Integer = viewInfo.ViewRects.Rows.Height - minRowHeigth
			Dim targetHeight As Integer = totalHeight / gridView1.DataRowCount
			e.RowHeight = Math.Max(minRowHeigth, targetHeight)
		End Sub
	End Class
End Namespace